src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBarSetReminder.vue

```diff
@@ -0,0 +1,244 @@
+		 */
+		id: {
+			type: [String, Number],
+			required: true,
+		},
+	},
+
+	data() {
+		return {
+			customDateTime: null,
+		}
+	},
+
+	computed: {
+		containerElement() {
+			return document.querySelector('.messages-list__scroller')
+		},
+
+		reminderOptions() {
+			const currentDateTime = moment()
+
+			// Same day 18:00 PM (or hidden)
+			const laterTodayTime = (currentDateTime.hour() < 18)
+				? moment().hour(18)
+				: null
+
+			// Tomorrow 08:00 AM
+			const tomorrowTime = moment().add(1, 'days').hour(8)
+
+			// Saturday 08:00 AM (or hidden)
+			const thisWeekendTime = (currentDateTime.day() !== 6 && currentDateTime.day() !== 0)
+				? moment().day(6).hour(8)
+				: null
+
+			// Next Monday 08:00 AM
+			const nextWeekTime = moment().add(1, 'weeks').day(1).hour(8)
+
+			return [
+				{
+					key: 'laterToday',
+					icon: Update,
+					timestamp: this.getTimestamp(laterTodayTime),
+					label: t('spreed', 'Later today — {timeLocale}', { timeLocale: laterTodayTime?.format('ddd LT') }),
+					ariaLabel: t('spreed', 'Set reminder for later today'),
+				},
+				{
+					key: 'tomorrow',
+					icon: ChevronRight,
+					timestamp: this.getTimestamp(tomorrowTime),
+					label: t('spreed', 'Tomorrow — {timeLocale}', { timeLocale: tomorrowTime?.format('ddd LT') }),
+					ariaLabel: t('spreed', 'Set reminder for tomorrow'),
+				},
+				{
+					key: 'thisWeekend',
+					icon: CalendarWeekend,
+					timestamp: this.getTimestamp(thisWeekendTime),
+					label: t('spreed', 'This weekend — {timeLocale}', { timeLocale: thisWeekendTime?.format('ddd LT') }),
+					ariaLabel: t('spreed', 'Set reminder for this weekend'),
+				},
+				{
+					key: 'nextWeek',
+					icon: ChevronDoubleRight,
+					timestamp: this.getTimestamp(nextWeekTime),
+					label: t('spreed', 'Next week — {timeLocale}', { timeLocale: nextWeekTime?.format('ddd LT') }),
+					ariaLabel: t('spreed', 'Set reminder for next week'),
+				},
+			].filter(option => option.timestamp !== null)
+		},
+	},
+
+	methods: {
+		getTimestamp(momentObject) {
+			return momentObject?.minute(0).second(0).millisecond(0).valueOf() || null
+		},
+
+		async setReminder(timestamp) {
+			try {
+				await setMessageReminder(this.token, this.id, timestamp / 1000)
+				showSuccess(t('spreed', 'A reminder was successfully set at {datetime}', {
+					datetime: moment(timestamp).format('LLL'),
+				}))
+			} catch (error) {
+				showError(t('spreed', 'Error occurred when creating a reminder'))
+			}
+		},
+
+		disableDatesBeforeToday(date) {
+			return date < moment().hour(0).minute(0).second(0).millisecond(0)
+		},
+
+		disableTimeBeforeNow(date) {
+			return date < moment()
+		},
+
+		setCustomReminder() {
+			this.setReminder(new Date(this.customDateTime).valueOf())
+
+			// Force close NcActions menu
+			let parent = this.$parent
+			while (parent) {
+				if (parent.$options.name === 'NcActions') {
+					parent.closeMenu(false)
+					return
+				}
+				parent = parent.$parent
+			}
+		},
+	},
```

Comment:
I'd highly prefer to avoid this way to close the menu...
- `closeMenu` is not mentioned as a public method of `NcActions`, I'd consider it to be an internal method
- A proper way to control `NcActions` open state is the `open` prop
- Iterating parents is also not a vue way to communicate between component 
- And `$parent` is available only by internal undocumented functions and properties in Composition API, it is not a recommended way to work with components 

If we do want to close the menu here, we can provide a method from `NcActions`'s parent component.

Reply:
And it doesn't work for me...
![1](https://github.com/nextcloud/spreed/assets/25978914/a589cb63-efb9-4b14-9310-d18201486556)

Reply:
Do we need to close the menu after setting a custom reminder? It is closed on hover loose. And if we do, should be also close on other options?

Reply:
IMO:
Close on set: yes 
Closed on hover loose: no